WooCommerce/Classes/ViewRelated/Search/SearchViewController.swift

```diff
@@ -417,16 +410,64 @@ extension SearchViewController {
 //
 private extension SearchViewController {
 
-    /// Displays the Empty State Legend.
+    /// Displays the view for the empty state.
     ///
     func displayEmptyState() {
-        emptyStateLabel.isHidden = false
+        // Create the controller if it doesn't exist yet
+        let childController: Command.EmptyStateViewControllerType = {
+            if let existing = emptyStateViewController {
+                return existing
+            } else {
+                let created = searchUICommand.createEmptyStateViewController()
+                emptyStateViewController = created
+                return created
+            }
+        }()
+        // Abort if we are already displaying this childController
+        guard childController.parent == nil else {
+            return
+        }
+
+        // Before creating the view (below), give the childController the keyboard adjustments
+        // they should use. This simplifies any keyboard observation they have in  `viewDidLoad`.
+        applyAdditionalKeyboardFrameHeightTo([childController])
+
+        // Create the view by accessing `.view`. This should trigger `viewDidLoad`.
+        guard let childView = childController.view else {
+            return
+        }
+
+        searchUICommand.configureEmptyStateViewControllerBeforeDisplay(viewController: childController,
+                                                                       searchKeyword: keyword)
+
+        childView.translatesAutoresizingMaskIntoConstraints = false
+
+        add(childController)
+        view.addSubview(childView)
+
+        // Match the position and size to the `tableView`. Attach top to the searchBar
+        NSLayoutConstraint.activate([
+            childView.leadingAnchor.constraint(equalTo: tableView.leadingAnchor),
+            childView.trailingAnchor.constraint(equalTo: tableView.trailingAnchor),
+            childView.topAnchor.constraint(equalTo: searchBar.bottomAnchor),
+            childView.bottomAnchor.constraint(equalTo: tableView.bottomAnchor)
+        ])
+
+        childController.didMove(toParent: self)
     }
 
-    /// Removes the Empty State Legend.
+    /// Removes the view for the empty state.
     ///
     func removeEmptyState() {
-        emptyStateLabel.isHidden = true
+        guard let childController = emptyStateViewController,
+              let childView = childController.view,
+              childController.parent == self else {
+            return
+        }
+
+        childController.willMove(toParent: nil)
+        childView.removeFromSuperview()
+        childController.removeFromParent()
```

Comment:
Like the `add` method, that `remove()` also has an odd sequence. The `removeFromParent()` is called **before** the `removeFromSuperview()`. No matter what reference I look at, like [this one](https://www.hackingwithswift.com/example-code/uikit/how-to-use-view-controller-containment) or [this one from Apple](https://developer.apple.com/library/archive/featuredarticles/ViewControllerPGforiPhoneOS/ImplementingaContainerViewController.html#//apple_ref/doc/uid/TP40007457-CH11-SW15), they say that the `removeFromParent()` should be called **after** the subview removal. 

This probably doesn't matter. But to avoid bugs, I think I'll keep this for now until we can make our own helpers that follow the Apple guidelines. 🙂